refinery/core/test_views.py

```diff
@@ -1922,56 +1922,105 @@ def setUp(self, **kwargs):
             view=SiteProfileAPIView.as_view()
         )
         self.current_site = Site.objects.get_current()
-        self.site_profile = SiteProfile.objects.create(
+        self.other_site = Site.objects.create(domain='google.com', name='test')
+        self.site_profile_current = SiteProfile.objects.create(
             site=self.current_site,
             about_markdown='About the platform paragraph.',
             intro_markdown='The refinery platform intro paragraph.',
             twitter_username='Mock_twitter_name'
         )
+        self.site_profile_other = SiteProfile.objects.create(
+            site=self.other_site,
+            about_markdown='About the other platform paragraph.',
+            intro_markdown='The other platform intro paragraph.',
+            twitter_username='Mock_twitter_name'
+        )
         self.site_video_1 = SiteVideo.objects.create(
             caption="Dashboard video",
-            site_profile=self.site_profile,
+            site_profile=self.site_profile_current,
             source="YouTube",
             source_id="yt_5tc"
         )
         self.site_video_2 = SiteVideo.objects.create(
             caption="Analysis video",
-            site_profile=self.site_profile,
+            site_profile=self.site_profile_current,
             source="YouTube",
             source_id="yt_875"
         )
 
         username = password = "admin"
         self.admin_user = User.objects.create_superuser(username, '', password)
-        self.get_request = self.factory.get(self.url_root)
+        self.get_request_current_true = self.factory.get(
+            self.url_root + '?current_site=true'
+        )
+        self.get_request_current_false = self.factory.get(
+            self.url_root + '?current_site=false'
+        )
+        self.get_request_current_foo = self.factory.get(
+            self.url_root + '?current_site=foo'
+        )
+        self.get_request_no_params = self.factory.get(self.url_root)
 
-    def test_get_returns_404_status_for_missing_site_profiles(self):
+    def test_serializer_list_empty(self):
         SiteProfile.objects.all().delete()
-        get_response = self.view(self.get_request)
+        self.assertEqual(
+            len(SiteProfileAPIView.all_site_profiles_serialized().data), 0
+        )
+
+    def test_serializer_list(self):
```

Comment:
* Noted below, method not necessary. *
1) More descriptive test function names.
2) Don't hardcode the length in case changes in set-up occurs.